File: 8/src/part1.rb
input = File.new("input.txt", "r")

counts = Array.new(10, 0)

input.each {
  |line|
  lineSegments = line.split(" | ")
  keySegment = lineSegments[1]
  signals = keySegment.split(" ")

  signals.each {
    |n|
    counts[n.length] = (counts[n.length] + 1)
  }
}

print "Part 1 answer: #{counts[2] + counts[3] + counts[4] + counts[7]}\n"

File: 10/src/part1.R
library("stringr")

processFile = function (filepath) {
  con = file(filepath, "r")
  points = 0

  while (TRUE) {
    line = readLines(con, n = 1)
    if (length(line) == 0) {
      break
    }

    newLine = line
    for (i in 1: 1000) {
      newLine = str_replace_all(newLine, "\\[\\]", "")
      newLine = str_replace_all(newLine, "\\(\\)", "")
      newLine = str_replace_all(newLine, "\\{\\}", "")
      newLine = str_replace_all(newLine, "\\<\\>", "")
    }

    if (str_detect(newLine, "\\[\\>|\\(\\>|\\{\\>")) {
      points = points + 25137
    } else if (str_detect(newLine, "\\[\\)|\\<\\)|\\{\\)")) {
      points = points + 3
    } else if (str_detect(newLine, "\\(\\]|\\<\\]|\\{\\]")) {
      points = points + 57
    } else if (str_detect(newLine, "\\[\\}|\\<\\}|\\(\\}")) {
      points = points + 1197
    }
  }

  print(paste("Part 1 answer: ", points))
  close(con)
}

processFile("input.txt")
